src/main/java/Nick.java

```diff
@@ -0,0 +1,142 @@
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Scanner;
+import java.util.logging.Logger;
+
+public class Nick {
+    public static int taskCount = 0;
+    public static final int TO_OFFSET_IDX = 4;
+    public static final int FROM_OFFSET_IDX = 6;
+    public static final int DEADLINE_OFFSET_IDX = 4;
+    private static final Logger LOGGER = Logger.getLogger(Nick.class.getName());
+
+    public static void main(String[] args) {
+        printIntroName();
+        printIntroMsg();
+
+        boolean hasQuit = false;
+        String command, commandAction;
+        Task[] userTasks = new Task[100];
+        Scanner input = new Scanner(System.in);
+
+        while (input.hasNextLine()) {
+            command = input.nextLine();
+            commandAction = command.split(" ")[0];
+
+            switch (commandAction) {
+            case "bye":
+            case "list":
+            case "mark":
+            case "unmark":
+                boolean continueProgram = continueExecution(command, taskCount, userTasks);
+                if (continueProgram) {
+                    continue;
+                }
+                hasQuit = true;
+            }
+            if (hasQuit) {
+                break;
+            }
+
+            taskExecution(command, commandAction, userTasks);
+        }
+    }
+
+    public static void taskExecution(String command, String taskType, Task[] userTasks) {
+        Task task;
+        String taskName;
+        int taskDescriptionIndex = taskType.length() + 1;
+        int taskDescriptionEndIndex = command.indexOf("/") - 1;
+
+        switch (taskType) {
+        case "todo":
+            taskName = command.substring(taskDescriptionIndex);
+            task = new Todo(taskName);
+            userTasks[taskCount] = task;
+            break;
+        case "deadline":
+            int deadlineIndex = command.indexOf("/by") + DEADLINE_OFFSET_IDX;
+            taskName = command.substring(taskDescriptionIndex, taskDescriptionEndIndex);
+            String deadline = command.substring(deadlineIndex);
+            task = new Deadline(taskName, deadline);
+            userTasks[taskCount] = task;
+            break;
+        case "event":
+            int fromIndex = command.indexOf("/from");
+            int toIndex = command.indexOf("/to");
+            taskName = command.substring(taskDescriptionIndex, taskDescriptionEndIndex);
+            String from = command.substring(fromIndex + FROM_OFFSET_IDX, toIndex - 1);
+            String to = command.substring(toIndex + TO_OFFSET_IDX);
+            task = new Event(taskName, from, to);
+            userTasks[taskCount] = task;
```

Comment:
I would  consider extracting codes in cases to a single method

Reply:
Good recommendation! Thank you.